lib/view_component/test_helpers.rb

```diff
@@ -162,5 +173,71 @@ def with_request_url(path)
     def build_controller(klass)
       klass.new.tap { |c| c.request = request }.extend(Rails.application.routes.url_helpers)
     end
+
+    # This is a simpler version of {Capybara::Session}.
+    #
+    # It only includes {Capybara::Node::Finders}, {Capybara::Node::Matchers},
+    # {#within} and {#within_element}. It is useful in that it does not require a
+    # session, an application or a driver, but can still use Capybara's finders
+    # and matchers on any string that contains HTML.
+    #
+    # Most of this code is shamelessly stolen from the {Capybara::Session} class
+    # in the Capybara gem (https://github.com/teamcapybara/capybara/blob/e704d00879fb1d1e1a0cc01e04c101bcd8af4a68/lib/capybara/session.rb#L38).
+    class SimpleSession
+      # Stolen from: https://github.com/teamcapybara/capybara/blob/e704d00879fb1d1e1a0cc01e04c101bcd8af4a68/lib/capybara/session.rb#L767-L774.
+
+      Capybara::Session::NODE_METHODS.each do |method|
+        if RUBY_VERSION >= "2.7"
+          class_eval <<~METHOD, __FILE__, __LINE__ + 1
+            def #{method}(...)
```

Comment:
Honestly I haven't given it much thought. It's copied from Capybara, as the comment indicates.

Reply:
I see, but we will need to maintain this code if it becomes part of ViewComponent so I feel it's important we understand the code.

Looking at the [git blame](https://github.com/teamcapybara/capybara/blame/e704d00879fb1d1e1a0cc01e04c101bcd8af4a68/lib/capybara/session.rb#L783-L793) for that particular section of code, it looks like it was added to address 2.7 compatibility issues, which seems related to the comment below. Since 2.7 introduced `...` and the keyword arguments change, we can avoid having to call `ruby2_keywords` and can use the `...` method of forwarding.